forge/csrc/passes/commute_utils.cpp

```diff
@@ -898,16 +898,195 @@ bool can_commute_past_op(
     return (is_elementwise(op) and op->op_name() != "interleave") or is_quantization_ops(op);
 }
 
+void update_select_attr(
+    graphlib::OpNode *select_op,
+    int select_dim,
+    std::optional<int> begin,
+    std::optional<int> length,
+    std::optional<int> stride)
+{
+    log_trace("op name = {}", select_op->name());
+
+    if (select_op->op_name() != "select")
+    {
+        log_warning("update_select_attr called for a non-select operation: {}", select_op->op_name());
+        return;
+    }
+
+    graphlib::OpType::Attrs named_attrs = select_op->named_attrs();
+    std::vector<graphlib::OpType::Attr> attr = select_op->op_attrs();
+    log_trace("current_attrs size = {}", attr.size());
+
+    if (attr.size() != 4)
+    {
+        log_error("Invalid attribute size for select operation: expected 4, got {}", attr.size());
+        return;
+    }
+
+    std::get<int>(attr[0]) = select_dim;
+    named_attrs["select_dim"] = select_dim;
+
+    if (begin.has_value())
+    {
+        std::get<int>(attr[1]) = begin.value();
+        named_attrs["begin"] = begin.value();
+    }
+    else
+    {
+        named_attrs["begin"] = std::get<int>(attr[1]);
+    }
+
+    if (length.has_value())
+    {
+        std::get<int>(attr[2]) = length.value();
+        named_attrs["length"] = length.value();
+    }
+    else
+    {
+        named_attrs["length"] = std::get<int>(attr[2]);
+    }
+
+    if (stride.has_value())
+    {
+        std::get<int>(attr[3]) = stride.value();
+        named_attrs["stride"] = stride.value();
+    }
+    else
+    {
+        named_attrs["stride"] = std::get<int>(attr[3]);
+    }
+
+    select_op->overwrite_op_named_attrs(attr, named_attrs);
+    log_trace(
+        "Updated select operation {}: select_dim = {}, begin = {}, length = {}, stride = {}",
+        select_op->name(),
+        select_dim,
+        begin.value_or(std::get<int>(attr[1])),
+        length.value_or(std::get<int>(attr[2])),
+        stride.value_or(std::get<int>(attr[3])));
+}
+
+void update_concat_attr(graphlib::OpNode *concatenate, int dim)
+{
+    if (concatenate->op_name() != "concatenate")
+        return;
+
+    std::vector<graphlib::OpType::Attr> attr;
+    attr.push_back(dim);
+
+    graphlib::OpType::Attrs named_attrs = concatenate->named_attrs();
+    named_attrs["dim"] = dim;
+
+    concatenate->overwrite_op_named_attrs(attr, named_attrs);
+    log_trace("Concatenate operation updated with new dim: {}", dim);
+}
+
+void update_vstack_attr(graphlib::OpNode *vstack, int slice_size)
+{
+    if (vstack->op_name() != "vstack")
```

Comment:
FYI, vstack will be deprecated eventually on FFE. However, if it's needed right now for cleaner structure I'm okay with pushing this through